Build error: specialization in different namespace

Hi, i am trying to compile cyclonedds-cxx 0.9.1 and got the following compile error:

It seems that the compiler does not like that `org::eclipse::cyclonedds::topic::TopicTraits::getSerType(allowable_encodings_t kinds = allowableEncodings())` is specialized in different namespaces (see log below).

## Error Log
```
[settings]
arch=armv8
arch_build=x86_64
build_type=Release
compiler=qcc
compiler.libcxx=cxx
compiler.version=5.4
os=Neutrino
os.version=7.0
os_build=Linux

[options]
cyclonedds_cxx:BUILD_IDLLIB=False
cyclonedds:BUILD_EXAMPLES=False
cyclonedds:BUILD_IDLC=False
cyclonedds:BUILD_TESTING=False
cyclonedds:ENABLE_IPV6=False
cyclonedds:ENABLE_SOURCE_SPECIFIC_MULTICAST=False

[2022-05-29T10:13:40.840Z] cyclonedds_cxx/0.9.1@jenkins/stable: Calling build()
[2022-05-29T10:13:40.840Z] -- The C compiler identification is QCC 5.4.0
[2022-05-29T10:13:40.840Z] -- The CXX compiler identification is QCC 5.4.0
[2022-05-29T10:13:40.840Z] -- Detecting C compiler ABI info
[2022-05-29T10:13:41.150Z] -- Detecting C compiler ABI info - done
[2022-05-29T10:13:41.414Z] -- Check for working C compiler: /data/jenkins/qnx700/host/linux/x86_64/usr/bin/qcc - skipped
[2022-05-29T10:13:41.414Z] -- Detecting C compile features
[2022-05-29T10:13:41.414Z] -- Detecting C compile features - done
[2022-05-29T10:13:41.414Z] -- Detecting CXX compiler ABI info
[2022-05-29T10:13:41.414Z] -- Detecting CXX compiler ABI info - done
[2022-05-29T10:13:41.414Z] -- Check for working CXX compiler: /data/jenkins/qnx700/host/linux/x86_64/usr/bin/q++ - skipped
[2022-05-29T10:13:41.414Z] -- Detecting CXX compile features
[2022-05-29T10:13:41.414Z] -- Detecting CXX compile features - done
[2022-05-29T10:13:41.414Z] -- Compiling with type discovery support
[2022-05-29T10:13:41.414Z] -- Compiling with topic discovery support
[2022-05-29T10:13:41.414Z] -- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY
[2022-05-29T10:13:41.415Z] -- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY - Success
[2022-05-29T10:13:41.415Z] -- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY
[2022-05-29T10:13:41.415Z] -- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY - Success
[2022-05-29T10:13:41.415Z] -- Performing Test COMPILER_HAS_DEPRECATED_ATTR
[2022-05-29T10:13:41.415Z] -- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
[2022-05-29T10:13:41.713Z] -- Configuring done
[2022-05-29T10:13:41.713Z] -- Generating done
[2022-05-29T10:13:41.713Z] CMake Warning:
[2022-05-29T10:13:41.713Z]   Manually-specified variables were not used by the project:
[2022-05-29T10:13:41.713Z] 
[2022-05-29T10:13:41.713Z]     CMAKE_EXPORT_NO_PACKAGE_REGISTRY
[2022-05-29T10:13:41.713Z]     CONAN_COMPILER
[2022-05-29T10:13:41.713Z]     CONAN_COMPILER_VERSION
[2022-05-29T10:13:41.713Z]     CONAN_EXPORTED
[2022-05-29T10:13:41.713Z]     CONAN_IN_LOCAL_CACHE
[2022-05-29T10:13:41.713Z]     CONAN_LIBCXX
[2022-05-29T10:13:41.713Z] 
[2022-05-29T10:13:41.713Z] 
[2022-05-29T10:13:41.713Z] -- Build files have been written to: /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/build_subfolder
[2022-05-29T10:13:41.713Z] [  3%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/dds/core/Duration.cpp.o
[2022-05-29T10:13:41.713Z] [  3%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/dds/core/Exception.cpp.o
[2022-05-29T10:13:41.713Z] [  5%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/dds/core/Reference.cpp.o
[2022-05-29T10:13:41.713Z] [  7%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/dds/core/policy/CorePolicy.cpp.o
[2022-05-29T10:13:41.713Z] [  8%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/dds/core/Time.cpp.o
[2022-05-29T10:13:41.713Z] [ 10%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/dds/core/status/State.cpp.o
[2022-05-29T10:13:42.012Z] [ 12%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/dds/domain/discovery.cpp.o
[2022-05-29T10:13:42.012Z] [ 14%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/dds/domain/find.cpp.o
[2022-05-29T10:13:42.012Z] [ 15%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/dds/pub/pubdiscovery.cpp.o
[2022-05-29T10:13:42.303Z] [ 17%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/dds/sub/subdiscovery.cpp.o
[2022-05-29T10:13:42.603Z] [ 19%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/dds/sub/subfind.cpp.o
[2022-05-29T10:13:42.603Z] [ 21%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/dds/sub/status/DataState.cpp.o
[2022-05-29T10:13:43.653Z] [ 22%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/core/Mutex.cpp.o
[2022-05-29T10:13:43.984Z] [ 24%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/core/ObjectDelegate.cpp.o
[2022-05-29T10:13:43.984Z] [ 26%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/core/DDScObjectDelegate.cpp.o
[2022-05-29T10:13:43.984Z] [ 28%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/core/ObjectSet.cpp.o
[2022-05-29T10:13:44.635Z] [ 29%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/core/EntityDelegate.cpp.o
[2022-05-29T10:13:44.921Z] [ 31%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/core/ReportUtils.cpp.o
[2022-05-29T10:13:45.253Z] [ 33%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/core/ListenerDispatcher.cpp.o
[2022-05-29T10:13:45.588Z] [ 35%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/core/InstanceHandleDelegate.cpp.o
[2022-05-29T10:13:45.589Z] [ 36%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/core/EntitySet.cpp.o
[2022-05-29T10:13:45.589Z] [ 38%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/core/MiscUtils.cpp.o
[2022-05-29T10:13:46.662Z] [ 40%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/core/cdr/fragchain.cpp.o
[2022-05-29T10:13:46.955Z] [ 42%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/core/cdr/cdr_stream.cpp.o
[2022-05-29T10:13:47.473Z] [ 43%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/core/cdr/basic_cdr_ser.cpp.o
[2022-05-29T10:13:47.746Z] [ 45%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/core/cdr/entity_properties.cpp.o
[2022-05-29T10:13:48.048Z] [ 47%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/core/cdr/extended_cdr_v1_ser.cpp.o
[2022-05-29T10:13:48.048Z] [ 49%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/core/cdr/extended_cdr_v2_ser.cpp.o
[2022-05-29T10:13:48.736Z] [ 50%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/core/cond/ConditionDelegate.cpp.o
[2022-05-29T10:13:49.353Z] [ 52%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/core/cond/GuardConditionDelegate.cpp.o
[2022-05-29T10:13:50.244Z] [ 54%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/core/cond/StatusConditionDelegate.cpp.o
[2022-05-29T10:13:50.244Z] [ 56%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/core/cond/WaitSetDelegate.cpp.o
[2022-05-29T10:13:50.539Z] [ 57%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/core/policy/PolicyDelegate.cpp.o
[2022-05-29T10:13:50.840Z] [ 59%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/domain/Domain.cpp.o
[2022-05-29T10:13:51.133Z] [ 61%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/domain/DomainWrap.cpp.o
[2022-05-29T10:13:51.133Z] [ 63%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/domain/DomainParticipantDelegate.cpp.o
[2022-05-29T10:13:51.772Z] [ 64%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/domain/DomainParticipantRegistry.cpp.o
[2022-05-29T10:13:52.034Z] [ 66%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/domain/qos/DomainParticipantQosDelegate.cpp.o
[2022-05-29T10:13:52.310Z] [ 68%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/pub/AnyDataWriterDelegate.cpp.o
[2022-05-29T10:13:53.738Z] [ 70%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/pub/PublisherDelegate.cpp.o
[2022-05-29T10:13:53.739Z] [ 71%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/pub/qos/DataWriterQosDelegate.cpp.o
[2022-05-29T10:13:53.739Z] [ 73%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/pub/qos/PublisherQosDelegate.cpp.o
[2022-05-29T10:13:54.042Z] [ 75%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/sub/qos/DataReaderQosDelegate.cpp.o
[2022-05-29T10:13:54.717Z] [ 77%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/sub/AnyDataReaderDelegate.cpp.o
[2022-05-29T10:13:55.053Z] [ 78%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/sub/SubscriberDelegate.cpp.o
[2022-05-29T10:13:55.054Z] [ 80%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/sub/BuiltinSubscriberDelegate.cpp.o
[2022-05-29T10:13:55.398Z] [ 82%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/sub/QueryDelegate.cpp.o
[2022-05-29T10:13:55.713Z] [ 84%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/sub/cond/ReadConditionDelegate.cpp.o
[2022-05-29T10:13:56.028Z] [ 85%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/sub/cond/QueryConditionDelegate.cpp.o
[2022-05-29T10:13:56.735Z] /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/src/org/eclipse/cyclonedds/sub/BuiltinSubscriberDelegate.cpp:131:134: error: specialization of 'template<class TOPIC> static ddsi_sertype* org::eclipse::cyclonedds::topic::TopicTraits<TOPIC>::getSerType(org::eclipse::cyclonedds::core::cdr::allowable_encodings_t)' in different namespace [-fpermissive]
[2022-05-29T10:13:56.735Z]  ddsi_sertype *org::eclipse::cyclonedds::topic::TopicTraits<dds::topic::ParticipantBuiltinTopicData>::getSerType(allowable_encodings_t)
[2022-05-29T10:13:56.735Z]                                                                                                                                       ^
[2022-05-29T10:13:56.735Z] In file included from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/org/eclipse/cyclonedds/topic/AnyTopicDelegate.hpp:23:0,
[2022-05-29T10:13:56.735Z]                  from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/dds/topic/detail/AnyTopic.hpp:22,
[2022-05-29T10:13:56.735Z]                  from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/dds/topic/AnyTopic.hpp:22,
[2022-05-29T10:13:56.735Z]                  from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/dds/topic/detail/Topic.hpp:27,
[2022-05-29T10:13:56.735Z]                  from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/dds/topic/Topic.hpp:21,
[2022-05-29T10:13:56.735Z]                  from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/dds/sub/detail/DataReader.hpp:15,
[2022-05-29T10:13:56.735Z]                  from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/dds/sub/DataReader.hpp:21,
[2022-05-29T10:13:56.735Z]                  from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/src/org/eclipse/cyclonedds/sub/BuiltinSubscriberDelegate.cpp:18:
[2022-05-29T10:13:56.735Z] /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/org/eclipse/cyclonedds/topic/TopicTraits.hpp:95:26: error:   from definition of 'template<class TOPIC> static ddsi_sertype* org::eclipse::cyclonedds::topic::TopicTraits<TOPIC>::getSerType(org::eclipse::cyclonedds::core::cdr::allowable_encodings_t)' [-fpermissive]
[2022-05-29T10:13:56.735Z]      static ddsi_sertype *getSerType(allowable_encodings_t kinds = allowableEncodings())
[2022-05-29T10:13:56.735Z]                           ^
[2022-05-29T10:13:56.735Z] /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/src/org/eclipse/cyclonedds/sub/BuiltinSubscriberDelegate.cpp:137:128: error: specialization of 'template<class TOPIC> static ddsi_sertype* org::eclipse::cyclonedds::topic::TopicTraits<TOPIC>::getSerType(org::eclipse::cyclonedds::core::cdr::allowable_encodings_t)' in different namespace [-fpermissive]
[2022-05-29T10:13:56.735Z]  ddsi_sertype *org::eclipse::cyclonedds::topic::TopicTraits<dds::topic::TopicBuiltinTopicData>::getSerType(allowable_encodings_t)
[2022-05-29T10:13:56.735Z]                                                                                                                                 ^
[2022-05-29T10:13:56.735Z] In file included from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/org/eclipse/cyclonedds/topic/AnyTopicDelegate.hpp:23:0,
[2022-05-29T10:13:56.735Z]                  from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/dds/topic/detail/AnyTopic.hpp:22,
[2022-05-29T10:13:56.735Z]                  from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/dds/topic/AnyTopic.hpp:22,
[2022-05-29T10:13:56.735Z]                  from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/dds/topic/detail/Topic.hpp:27,
[2022-05-29T10:13:56.735Z]                  from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/dds/topic/Topic.hpp:21,
[2022-05-29T10:13:56.735Z]                  from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/dds/sub/detail/DataReader.hpp:15,
[2022-05-29T10:13:56.735Z]                  from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/dds/sub/DataReader.hpp:21,
[2022-05-29T10:13:56.735Z]                  from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/src/org/eclipse/cyclonedds/sub/BuiltinSubscriberDelegate.cpp:18:
[2022-05-29T10:13:56.735Z] /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/org/eclipse/cyclonedds/topic/TopicTraits.hpp:95:26: error:   from definition of 'template<class TOPIC> static ddsi_sertype* org::eclipse::cyclonedds::topic::TopicTraits<TOPIC>::getSerType(org::eclipse::cyclonedds::core::cdr::allowable_encodings_t)' [-fpermissive]
[2022-05-29T10:13:56.735Z]      static ddsi_sertype *getSerType(allowable_encodings_t kinds = allowableEncodings())
[2022-05-29T10:13:56.735Z]                           ^
[2022-05-29T10:13:56.735Z] /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/src/org/eclipse/cyclonedds/sub/BuiltinSubscriberDelegate.cpp:143:134: error: specialization of 'template<class TOPIC> static ddsi_sertype* org::eclipse::cyclonedds::topic::TopicTraits<TOPIC>::getSerType(org::eclipse::cyclonedds::core::cdr::allowable_encodings_t)' in different namespace [-fpermissive]
[2022-05-29T10:13:56.735Z]  ddsi_sertype *org::eclipse::cyclonedds::topic::TopicTraits<dds::topic::PublicationBuiltinTopicData>::getSerType(allowable_encodings_t)
[2022-05-29T10:13:56.735Z]                                                                                                                                       ^
[2022-05-29T10:13:56.736Z] In file included from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/org/eclipse/cyclonedds/topic/AnyTopicDelegate.hpp:23:0,
[2022-05-29T10:13:56.736Z]                  from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/dds/topic/detail/AnyTopic.hpp:22,
[2022-05-29T10:13:56.736Z]                  from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/dds/topic/AnyTopic.hpp:22,
[2022-05-29T10:13:56.736Z]                  from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/dds/topic/detail/Topic.hpp:27,
[2022-05-29T10:13:56.736Z]                  from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/dds/topic/Topic.hpp:21,
[2022-05-29T10:13:56.736Z]                  from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/dds/sub/detail/DataReader.hpp:15,
[2022-05-29T10:13:56.736Z]                  from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/dds/sub/DataReader.hpp:21,
[2022-05-29T10:13:56.736Z]                  from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/src/org/eclipse/cyclonedds/sub/BuiltinSubscriberDelegate.cpp:18:
[2022-05-29T10:13:56.736Z] /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/org/eclipse/cyclonedds/topic/TopicTraits.hpp:95:26: error:   from definition of 'template<class TOPIC> static ddsi_sertype* org::eclipse::cyclonedds::topic::TopicTraits<TOPIC>::getSerType(org::eclipse::cyclonedds::core::cdr::allowable_encodings_t)' [-fpermissive]
[2022-05-29T10:13:56.736Z]      static ddsi_sertype *getSerType(allowable_encodings_t kinds = allowableEncodings())
[2022-05-29T10:13:56.736Z]                           ^
[2022-05-29T10:13:56.736Z] /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/src/org/eclipse/cyclonedds/sub/BuiltinSubscriberDelegate.cpp:149:135: error: specialization of 'template<class TOPIC> static ddsi_sertype* org::eclipse::cyclonedds::topic::TopicTraits<TOPIC>::getSerType(org::eclipse::cyclonedds::core::cdr::allowable_encodings_t)' in different namespace [-fpermissive]
[2022-05-29T10:13:56.736Z]  ddsi_sertype *org::eclipse::cyclonedds::topic::TopicTraits<dds::topic::SubscriptionBuiltinTopicData>::getSerType(allowable_encodings_t)
[2022-05-29T10:13:56.736Z]                                                                                                                                        ^
[2022-05-29T10:13:56.736Z] In file included from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/org/eclipse/cyclonedds/topic/AnyTopicDelegate.hpp:23:0,
[2022-05-29T10:13:56.736Z]                  from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/dds/topic/detail/AnyTopic.hpp:22,
[2022-05-29T10:13:56.736Z]                  from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/dds/topic/AnyTopic.hpp:22,
[2022-05-29T10:13:56.736Z]                  from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/dds/topic/detail/Topic.hpp:27,
[2022-05-29T10:13:56.736Z]                  from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/dds/topic/Topic.hpp:21,
[2022-05-29T10:13:56.736Z]                  from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/dds/sub/detail/DataReader.hpp:15,
[2022-05-29T10:13:56.736Z]                  from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/dds/sub/DataReader.hpp:21,
[2022-05-29T10:13:56.736Z]                  from /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/src/org/eclipse/cyclonedds/sub/BuiltinSubscriberDelegate.cpp:18:
[2022-05-29T10:13:56.736Z] /data/jenkins/workspace/conan_cyclonedds_cxx_2/cnn_l_1/.conan/data/cyclonedds_cxx/0.9.1/jenkins/stable/build/c988340d9ca4f5397b1968890330364cf6923b38/source_subfolder/src/ddscxx/include/org/eclipse/cyclonedds/topic/TopicTraits.hpp:95:26: error:   from definition of 'template<class TOPIC> static ddsi_sertype* org::eclipse::cyclonedds::topic::TopicTraits<TOPIC>::getSerType(org::eclipse::cyclonedds::core::cdr::allowable_encodings_t)' [-fpermissive]
[2022-05-29T10:13:56.736Z]      static ddsi_sertype *getSerType(allowable_encodings_t kinds = allowableEncodings())
[2022-05-29T10:13:56.736Z]                           ^
[2022-05-29T10:13:56.736Z] [ 87%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/sub/qos/SubscriberQosDelegate.cpp.o
[2022-05-29T10:13:57.361Z] [ 89%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/topic/find.cpp.o
[2022-05-29T10:13:57.362Z] [ 91%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/topic/hash.cpp.o
[2022-05-29T10:13:57.362Z] [ 92%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/topic/AnyTopicDelegate.cpp.o
[2022-05-29T10:13:57.942Z] [ 94%] Building CXX object src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/topic/FilterDelegate.cpp.o
[2022-05-29T10:13:57.943Z] cc: /data/jenkins/qnx700/host/linux/x86_64/usr/lib/gcc/aarch64-unknown-nto-qnx7.0.0/5.4.0/cc1plus error 1
[2022-05-29T10:13:57.943Z] make[2]: *** [src/ddscxx/CMakeFiles/ddscxx.dir/build.make:706: src/ddscxx/CMakeFiles/ddscxx.dir/src/org/eclipse/cyclonedds/sub/BuiltinSubscriberDelegate.cpp.o] Error 1
[2022-05-29T10:13:57.943Z] make[2]: *** Waiting for unfinished jobs....
[2022-05-29T10:13:59.986Z] make[1]: *** [CMakeFiles/Makefile2:131: src/ddscxx/CMakeFiles/ddscxx.dir/all] Error 2
[2022-05-29T10:13:59.986Z] make: *** [Makefile:156: all] Error 2
```

--- a/src/ddscxx/src/org/eclipse/cyclonedds/sub/BuiltinSubscriberDelegate.cpp
+++ b/src/ddscxx/src/org/eclipse/cyclonedds/sub/BuiltinSubscriberDelegate.cpp
@@ -160,10 +160,10 @@ ddsi_sertype *org::eclipse::cyclonedds::topic::TopicTraits<dds::topic::Subscript
     return nullptr;
 }
 
-}
-}
-}
-}
+} // topic
+} // cyclonedds
+} // eclipse
+} // org
     
 org::eclipse::cyclonedds::sub::AnyDataReaderDelegate::ref_type
 org::eclipse::cyclonedds::sub::BuiltinSubscriberDelegate::get_builtin_reader(

--- a/src/ddscxx/src/org/eclipse/cyclonedds/sub/BuiltinSubscriberDelegate.cpp
+++ b/src/ddscxx/src/org/eclipse/cyclonedds/sub/BuiltinSubscriberDelegate.cpp
@@ -127,6 +127,15 @@ org::eclipse::cyclonedds::sub::create_builtin_reader(
     return reader;
 }
 
+namespace org
+{
+namespace eclipse
+{
+namespace cyclonedds
+{
+namespace topic
+{
+
 template<>
 ddsi_sertype *org::eclipse::cyclonedds::topic::TopicTraits<dds::topic::ParticipantBuiltinTopicData>::getSerType(allowable_encodings_t)
 {
@@ -151,6 +160,11 @@ ddsi_sertype *org::eclipse::cyclonedds::topic::TopicTraits<dds::topic::Subscript
     return nullptr;
 }
 
+}
+}
+}
+}
+    
 org::eclipse::cyclonedds::sub::AnyDataReaderDelegate::ref_type
 org::eclipse::cyclonedds::sub::BuiltinSubscriberDelegate::get_builtin_reader(
     SubscriberDelegate& subscriber,

--- a/src/ddscxx/src/org/eclipse/cyclonedds/sub/BuiltinSubscriberDelegate.cpp
+++ b/src/ddscxx/src/org/eclipse/cyclonedds/sub/BuiltinSubscriberDelegate.cpp
@@ -137,25 +137,25 @@ namespace topic
 {
 
 template<>
-ddsi_sertype *org::eclipse::cyclonedds::topic::TopicTraits<dds::topic::ParticipantBuiltinTopicData>::getSerType(allowable_encodings_t)
+ddsi_sertype *TopicTraits<dds::topic::ParticipantBuiltinTopicData>::getSerType(allowable_encodings_t)
 {
     return nullptr;
 }
 
 template<>
-ddsi_sertype *org::eclipse::cyclonedds::topic::TopicTraits<dds::topic::TopicBuiltinTopicData>::getSerType(allowable_encodings_t)
+ddsi_sertype *TopicTraits<dds::topic::TopicBuiltinTopicData>::getSerType(allowable_encodings_t)
 {
     return nullptr;
 }
 
 template<>
-ddsi_sertype *org::eclipse::cyclonedds::topic::TopicTraits<dds::topic::PublicationBuiltinTopicData>::getSerType(allowable_encodings_t)
+ddsi_sertype *TopicTraits<dds::topic::PublicationBuiltinTopicData>::getSerType(allowable_encodings_t)
 {
     return nullptr;
 }
 
 template<>
-ddsi_sertype *org::eclipse::cyclonedds::topic::TopicTraits<dds::topic::SubscriptionBuiltinTopicData>::getSerType(allowable_encodings_t)
+ddsi_sertype *TopicTraits<dds::topic::SubscriptionBuiltinTopicData>::getSerType(allowable_encodings_t)
 {
     return nullptr;
 }

